### ISSUE TITLE
SkiaSharp managed stream exception on MacOS

### ISSUE THREAD

**Issue by ScrubN** [lay295/TwitchDownloader#404]
### Checklist

- [X] I have checked the [issue page](https://github.com/lay295/TwitchDownloader/issues) for duplicates
- [X] I am running the latest version ([download here](https://github.com/lay295/TwitchDownloader/releases/latest))

### Describe your issue here

**This is a MacOS only issue!**

Downloading a chat with embeded emotes throws an exception within SkiaSharp:
```
$ ./TwitchDownloaderCLI -m ChatDownload -u [redacted for privacy] -o chat.json --embed-emotes
[STATUS] - Downloading + Embedding Emotes
Unhandled exception. System.AggregateException: One or more errors occurred. (The type initializer for ‘SkiaSharp.SKAbstractManagedStream' threw an exception.)
 ---> System.TypeInitializationException: The type initializer for 'SkiaSharp.SKAbstractManagedStream' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'libSkiaSharp' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(liblibSkiaSharp, 1): image not found
   at SkiaSharp.SkiaApi.sk_managedstream_set_procs(SKManagedStreamDelegates procs)
   at SkiaSharp.SKAbstractManagedStream..cctor()
   --- End of inner exception stack trace ---
   at SkiaSharp.SKAbstractManagedStream..ctor(Boolean owns)
   at SkiaSharp.SKManagedStream..ctor(Stream managedStream, Boolean disposeManagedStream)
   at SkiaSharp.SKCodec.WrapManagedStream(Stream stream)
   at SkiaSharp.SKCodec.Create(Stream stream, SKCodecResult& result)
   at SkiaSharp.SKCodec.Create(Stream stream)
   at TwitchDownloaderCore.TwitchObjects.TwitchEmote..ctor(Byte[] imageData, EmoteProvider emoteProvider, Int32 imageScale, String imageId, String imageName)
   at TwitchDownloaderCore.TwitchHelper.GetThirdPartyEmotes(Int32 streamerId, String cacheFolder, Emotes embededEmotes, Boolean bttv, Boolean ffz, Boolean stv)
   at TwitchDownloaderCore.ChatDownloader.DownloadAsync(IProgress`1 progress, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean )
   at System.Threading.Tasks.Task.Wait(Int32 , CancellationToken )
   at System.Threading.Tasks.Task.Wait()
   at TwitchDownloaderCLI.Program.DownloadChat(Options inputOptions)
   at TwitchDownloaderCLI.Program.Main(String[] args)
Abort trap: 6
```

I used tesseract to extract the stacktrace from a photo of the terminal on my mac. If there's any typos, that'll be why.

**Comment by ScrubN** [comment]
SkiaSharp is currently on 2.80.2 but can be updated to 2.88.3. It's possible that may fix it

**Comment by lay295** [comment]
Can give 1.50.6 a try. I couldn't find a nuget package version that matched the others so might be a problem? But probably not.

**Comment by ScrubN** [comment]
New error:

```
$ ./TwitchDownloaderCLI -m ChatDownload -u [redacted for privacy] -o chat.json --embed-emotes
[STATUS] - Downloading + Embedding Emotes
Unhandled exception. System.AggregateException: One or more errors occurred. (The type initializer for 'SkiaSharp.SKData' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'SkiaSharp.SKData' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'libSkiaSharp' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(liblibSkiaSharp, 1): image not found
   at SkiaSharp.SkiaApi.sk_data_new_empty()
   at SkiaSharp.SKData..cctor()
   --- End of inner exception stack trace ---
   at SkiaSharp.SKData.CreateCopy(Byte[] bytes)
   at SkiaSharp.SKImage.FromEncodedData(Byte[] data)
   at TwitchDownloaderCore.TwitchHelper.GetImage(String cachePath, String imageUrl, String imageId, String imageScale, String imageType)
   at TwitchDownloaderCore.TwitchHelper.GetThirdPartyEmotes(Int32 streamerId, String cacheFolder, Emotes embededEmotes, Boolean bttv, Boolean ffz, Boolean stv)
   at TwitchDownloaderCore.ChatDownloader.DownloadAsync(IProgress`1 progress, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean )
   at System.Threading.Tasks.Task.Wait(Int32 , CancellationToken )
   at System.Threading.Tasks.Task.Wait()
   at TwitchDownloaderCLI.Program.DownloadChat(Options inputOptions)
   at TwitchDownloaderCLI.Program.Main(String[] args)
Abort trap: 6
```

Wondering if the nuget packages being out of date is possible the primary problem

**Comment by ScrubN** [comment]
I figured out compiling for MacOS and found the issue...

When you compile, it produces the binary, 3 `.DYLIB` files, and 2 `.pdb` files. The binary need the dylibs to work.

**Comment by ScrubN** [comment]
Also while we're on the topic of NuGet packages, Newtonsoft.Json has a critical security fix, it should be updated to 13.0.1

**Comment by ScrubN** [comment]
fixed by https://github.com/lay295/TwitchDownloader/commit/cae9944710d26125c9cedac326f10668a75ca12f
